commonjs-extension-resolution-loader/README.md

```diff
@@ -4,7 +4,14 @@ of file extensions and the ability to import directories that have an index
 file.
 
 Use this loader to enable automatic extension resolution and importing from
-directories that include an index file.
+directories that include an index file, like this:
+
+```js
+import file from './file'; // Where ./file is ./file.js or ./file.mjs
+import index from './folder'; // Where ./folder is ./folder/index.js or ./folder/index.mjs
```

Comment:
I don’t think it automatically searches for `.mjs`. This is to match the behavior of the old flag.

Reply:
It works fine on my machine.

```console
$ git clean -xdf
$ npm i
$ echo "export default 'foo';" > foo.mjs
$ echo "import foo from './foo'; console.log({foo});" > test.mjs
$ node --version
v19.0.0
$ node --loader=./loader.js test.mjs
(node:7440) ExperimentalWarning: Custom ESM Loaders is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
{ foo: 'foo' }
```

Reply:
I don’t know why that’s happening. This test passes:

https://github.com/nodejs/loaders-test/blob/985a3c752e866f079d50f17ff1a52a8e00c672b7/commonjs-extension-resolution-loader/test/test-esm-specifiers.js#L40-L52

Reply:
Actually, `test/fixtures/es-module-specifiers/implicit-main-type-commonjs` doesn’t exist in any form, so that’s why it’s passing (testing for a “module not found” error). I’m fine with saying that the loader searches for `.mjs` extensions and we just delete this bad test.